Dockerfile

```diff
@@ -53,133 +91,220 @@ RUN --mount=type=cache,target=/tmp/.gradle/ \
+    && rm commandlinetools-linux-${ANDROID_SDK_VERSION}_latest.zip \
+    && PROXY_HOST_AND_PORT=${https_proxy#*://} \
+    && if [ -n "$PROXY_HOST_AND_PORT" ]; then \
+       # While sdkmanager uses HTTPS by default, the proxy type is still called "http".
+       SDK_MANAGER_PROXY_OPTIONS="--proxy=http --proxy_host=${PROXY_HOST_AND_PORT%:*} --proxy_port=${PROXY_HOST_AND_PORT##*:}"; \
+    fi \
+    && yes | $ANDROID_HOME/cmdline-tools/bin/sdkmanager $SDK_MANAGER_PROXY_OPTIONS \
+       --sdk_root=$ANDROID_HOME "platform-tools" "cmdline-tools;latest"
+COPY docker/android.sh /etc/profile.d
+
+#------------------------------------------------------------------------
+# SCANCODE - Build ScanCode as a separate component
+FROM pythonbuild AS scancodebuild
+
+ARG SCANCODE_VERSION="3.2.1rc2"
+ENV SCANCODE_URL "https://github.com/nexB/scancode-toolkit/archive/refs/tags/"
+ENV PATH=/opt/python/shims:$PATH
+
+RUN mkdir -p /opt/scancode \
+    && curl -ksSL ${SCANCODE_URL}/v${SCANCODE_VERSION}.tar.gz | tar -C /opt/scancode -xz --strip-components=1 \
+    && cd /opt/scancode \
+    && PYTHON_EXE=python3 ./configure \
+    # Clean up unneeded installed binaries
+    && rm -rf /opt/scancode/thirdparty
+
+#------------------------------------------------------------------------
+# Main container
+FROM eclipse-temurin:11-jre
+
+# ORT
+COPY --from=ortbuild /opt/ort /opt/ort
+COPY docker/ort-wrapper.sh /usr/bin/ort
+COPY docker/ort-wrapper.sh /usr/bin/orth
+RUN chmod 755 /usr/bin/ort
+
+# Python
+COPY --from=pythonbuild /opt/python /opt/python
+COPY --from=pythonbuild /etc/profile.d/python.sh /etc/profile.d/
+
+# Ruby
+COPY --from=rubybuild /opt/rbenv /opt/rbenv
+COPY --from=rubybuild /etc/profile.d/ruby.sh /etc/profile.d/
+
+# NodeJS
+COPY --from=nodebuild /opt/nodejs /opt/nodejs
+COPY --from=nodebuild /etc/profile.d/nodejs.sh /etc/profile.d/
+
+# Golang
+COPY --from=gobuild /opt/go /opt/go/
+COPY --from=gobuild  /etc/profile.d/go.sh /etc/profile.d/
+
+# Haskell
+COPY --from=haskellbuild /usr/bin/stack /usr/bin
+
+# Repo and Android
+COPY --from=androidbuild /usr/bin/repo /usr/bin/
+COPY --from=androidbuild /opt/android-sdk /opt/android-sdk
+COPY --from=androidbuild /etc/profile.d/android.sh /etc/profile.d/
+
+# ScanCode
+COPY --from=scancodebuild /opt/scancode /opt/scancode
+RUN ln -s /opt/scancode/bin/scancode /usr/bin/scancode \
+    && ln -s /opt/scancode/bin/pip /usr/bin/scancode-pip \
+    && ln -s /opt/scancode/bin/extractcode /usr/bin/extractcode
+
+RUN  --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/apt apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+        ca-certificates \
+        cvs \
+        curl \
+        gnupg \
+        libarchive-tools \
         netbase \
-        openssh-client \
-        unzip \
-        xz-utils \
-        zlib1g-dev \
-        # Install VCS tools (no specific versions required here).
+        openssl \
+        unzip
```

Comment:
no `&& rm -rf /var/lib/apt/lists/* ` here?

Reply:
Nope, we will need install another set of packages some lines below,m then this avoid a second update.
Second line removes the lists then.

Reply:
But that increases the size of the layer, and I thought it was best practice to always clean up.

Reply:
AFAIK every `RUN` / `ARG` / `ENV` results in its own layer.

Reply:
Solved